eva-lib/src/main/java/uk/ac/ebi/eva/lib/utils/RepositoryFilter.java

```diff
@@ -0,0 +1,139 @@
+package uk.ac.ebi.eva.lib.utils;
+
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import uk.ac.ebi.eva.lib.repository.VariantEntityRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class RepositoryFilter<T> {
+
+    private String field;
+    private T value;
+    private VariantEntityRepository.RelationalOperator operator = VariantEntityRepository.RelationalOperator.NONE;
+
+    public RepositoryFilter(String field, T value, VariantEntityRepository.RelationalOperator operator) {
+        this.field = field;
+        this.value = value;
+        this.operator = operator;
+    }
+
+    public Query apply(Query query) {
+        Criteria criteria = Criteria.where(field);
+
+        switch (operator) {
+            case EQ:
+                criteria = criteria.is(value);
+                break;
+            case GT:
+                criteria = criteria.gt(value);
+                break;
+            case LT:
+                criteria = criteria.lt(value);
+                break;
+            case GTE:
+                criteria = criteria.gte(value);
+                break;
+            case LTE:
+                criteria = criteria.lte(value);
+                break;
+            case IN:
+                criteria = criteria.in(value);
+                break;
+            case NONE:
+                throw new IllegalArgumentException();
+        }
+
+        query.addCriteria(criteria);
+        return query;
+    }
+
+    public static List<RepositoryFilter> getRepositoryFilters(String maf, String polyphenScore, String siftScore,
```

Comment:
With the model described in the comment and, given the real use case in the rest services, we know always this string should be the maf, this the polyphen score, so there shouldn't be any need to create a factory and the checkings could be put in the constructor of the classes.